src/git.ts

```diff
@@ -0,0 +1,122 @@
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import * as core from "@actions/core";
+import parseDiff, { File as DiffFile } from "parse-diff";
+import { ReviewComment } from "./types";
+
+const GITHUB_TOKEN = core.getInput("GITHUB_TOKEN");
+const octokit = new Octokit({ auth: GITHUB_TOKEN });
+
+export async function getDiff(
+  owner: string,
+  repo: string,
+  pull_number: number,
+  action: string
+) {
+  // Do a full diff on new PRs, but do an incremental diff on updates
+  if (action === "opened") {
```

Comment:
The `action` parameter is not type-checked. Consider using a union type like `'opened' | 'synchronize'` to ensure type safety and prevent potential runtime errors.